HACK.md

```diff
@@ -72,25 +72,37 @@ make test
+
+The following table contains a list of environment variables you that will override specific paths under the **Build** CRD.
 
 | Environment Variable               | Path                           | Description                                                   |
 |------------------------------------|--------------------------------|---------------------------------------------------------------|
-| `TEST_NAMESPACE`                   | _none_                         | Target namespace to execute tests upon, default is `default`. |
 | `TEST_IMAGE_REPO`                  | `spec.output.image`            | Image repository for end-to-end tests                         |
 | `TEST_IMAGE_REPO_SECRET`           | `spec.output.credentials.name` | Container credentials secret name                             |
 | `TEST_IMAGE_REPO_DOCKERCONFIGJSON` | _none_                         | JSON payload equivalent to `~/.docker/config.json`            |
-| `TEST_E2E_FLAGS`                   | _none_                         | Ginkgo flags.                                                 |
-
-The default for `TEST_E2E_FLAGS` is `-failFast -p -randomizeAllSpecs -slowSpecThreshold=300 -timeout=15m -trace -v`. See all Ginkgo flags here: [The Ginkgo CLI](https://onsi.github.io/ginkgo/#the-ginkgo-cli). Especially of interest are `--focus` and `--skip` to run selective tests.
 
 The contents of `TEST_IMAGE_REPO_DOCKERCONFIGJSON` can be obtained from [quay.io](quay.io) using a [robot account](https://docs.quay.io/glossary/robot-accounts.html). The JSON payload is for example:
 
 ```json
 { "auths": { "quay.io": { "auth": "<secret-credentials>" } } }
 ```
 
-When both `TEST_IMAGE_REPO_SECRET` and `TEST_IMAGE_REPO_DOCKERCONFIGJSON` are informed, a new secret is created for end-to-end tests, named by  `TEST_IMAGE_REPO_SECRET`. However, when `TEST_IMAGE_REPO_DOCKERCONFIGJSON` is empty, e2e tests are expecting to find a pre-existing one.
+When both `TEST_IMAGE_REPO_SECRET` and `TEST_IMAGE_REPO_DOCKERCONFIGJSON` are informed, a new secret is created for end-to-end tests, named by `TEST_IMAGE_REPO_SECRET`. However, when `TEST_IMAGE_REPO_DOCKERCONFIGJSON` is empty, e2e tests are expecting to find a pre-existing one.
+
+The following table contains a list of environment variables you that will override specific paths under the **BuildRun** CRD.
+
+| Environment Variable          | Path                   | Description |
+|--------------------------------|-----------------------|-------------|
+| `TEST_E2E_SERVICEACCOUNT_NAME` | `spec.serviceAccount` | The name of the service account used by the build runs, the code will try to create the service account but not fail if it already exists. Special value is generated which will lead to using the auto-generation feature for each build run. |
```

Comment:
Does `Special value is generated which will lead to using the auto-generation feature for each build run.` needs to be `Special value is "generated",  which will lead to using the auto-generation feature for each build run.` ?

Why you think we need this functionality?

Reply:
Yes, I think so. The goal of this PR is to allow running the e2e tests even if the user has not access to all k8s resources. If the user has no access to service accounts, the auto-generation is the only possible way, I think.

Reply:
Corrected the documentation.